### DISCUSSION TITLE
Timewarrior

### DISCUSSION THREAD

**Discussion by davidbgonz** [catppuccin/catppuccin#1948] [Port Requests]
**Are you available to help in the creation of this port? 😄 (optional)**
Yes

**Additional context 📷 (optional)**
Timewarrior is a companion to Taskwarrior created by the same team. They are independent packages with separate configs so I am creating a separate issue (and port) for the Timewarrior port.

**Comment by justTOBBI** [comment]
So after spending some time looking into this, it appears as if taskwarrior nor timewarrior can use custom defined colors and just use the terminal or system colors.

**Comment by davidbgonz** [comment]
I completely forgot about this. Yeah, when I had tried out a test theme it was all based on using the base colors of the terminal and defining them in a theme config.

Example snippet:
```
# General decoration
color.label=magenta
color.label.sort=magenta
color.alternate=
color.header=cyan
color.footnote=yellow
color.warning=bold red
color.error=white on red
color.debug=blue
```

So not really a catppuccin theme, and more of a base16 theme. The build-in `dark-16.theme` probably more than suffices for most people I guess.

**Comment by unseen-ninja** [comment]
A way around this could be to just reference the base16 colours we want and encourage the user to use a catppuccin colour scheme for their terminal?

**Comment by davidbgonz** [comment]
We could keep a baseline theme file for people to use/start out with, and in the readme explain that this works based on the terminal colors (then add a link to the section of terminal ports).

In each image, the top one is the example theme I made, the bottom one is dark-16. The `dark-16.theme` is pretty meh.
![image](https://user-images.githubusercontent.com/10837237/194347828-749112e0-cc74-4c9d-aa05-6899a261c0c6.png)
![image](https://user-images.githubusercontent.com/10837237/194348177-9d931033-9c03-4c2f-9b66-4763ef55d9f7.png)

**Comment by davidbgonz** [comment]
Side note: I'm pretty sure my terminal is still on ctp 1.0 so ignore any slight variations in color.

**Comment by unseen-ninja** [comment]
Tho, this causes issues in some places as seen below.
!()[https://user-images.githubusercontent.com/35840154/194351466-18920cd5-515e-404d-a396-358fc9b3d56e.png]
Defining the colour pairs would ensure that it works with the catppuccin colourscheme 100%, no? o:

**Comment by davidbgonz** [comment]
Yeah, in a theme file you can just overwrite all the colors either using the term color name or number. The test theme I created is essentially the dark-16 theme with adjustments made with the ctp colors in mind.

**Comment by unseen-ninja** [comment]
We should probably have a simple config then just to ensure we don't run into contrast issues. c:

**Comment by soupglasses** [comment]
@davidbgonz do you have your customized configuration anywhere as reference?

**Comment by davidbgonz** [comment]
I can throw my configs for both taskwarror and timewarror in a repo.

**Comment by mdoury** [comment]
> I can throw my configs for both taskwarror and timewarror in a repo.

@davidbgonz any updates on this?

**Comment by davidbgonz** [comment]
Thanks for the ping, this completely slipped my mind. For now I added the time/task warrior configs to my horrible excuse of a dotfiles repo: https://github.com/davidbgonz/dotfiles/tree/main/home
